compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -537,8 +529,14 @@ where
         frame: usize,
         local: mir::Local,
     ) -> InterpResult<'tcx, PlaceTy<'tcx, M::Provenance>> {
-        let layout = self.layout_of_local(&self.stack()[frame], local, None)?;
-        let place = Place::Local { frame, local, offset: None };
+        // Other parts of the system rely on `Place::Local` never being unsized.
+        // So we eagerly check here if this local has an MPlace, and if yes we use it.
+        let frame_ref = &self.stack()[frame];
+        let layout = self.layout_of_local(frame_ref, local, None)?;
+        let place = match frame_ref.locals[local].access()? {
```

Comment:
This seems to be a hot function so I think this extra check here is costing us. But I think this is cleaner than each `PlaceTy` method having to support the case of "it's a `Place::Local` tosomething unsized, let's get the underlying in-memory place before doing anything"...